sklearnex/covariance/incremental_covariance.py

```diff
@@ -197,6 +199,39 @@ def _onedal_partial_fit(self, X, queue=None, check_input=True):
 
         return self
 
+    @wrap_output_data
+    def score(self, X_test, y=None):
+        xp, _ = get_namespace(X_test)
+
+        location = self.location_
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(
+                X_test,
+                dtype=[np.float64, np.float32],
+                reset=False,
+                copy=self.copy,
+            )
+        else:
+            X = check_array(
+                X_test,
+                dtype=[np.float64, np.float32],
+                copy=self.copy,
+            )
+
+        if "numpy" not in xp.__name__:
+            location = xp.asarray(location, device=X_test.device)
+            if isinstance(X, np.ndarray):
+                X = X_test
```

Comment:
X=X_test is basically bypassing the check_array or _validate_data, but still the checks will be performed. Should the checks be conditional?

Reply:
I want the checks to always occur, but I want to prevent casting the dpnp/dpctl tensor to a numpy array using asarray() in them. Do you think I should add a comment in the code explaining this weirdness?

Reply:
I see, probably a comment will be better.

Reply:
Comment added